interfaces/polkit/spec.go

```diff
@@ -0,0 +1,106 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package polkit
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+type Policy []byte
+
+// Specification keeps all the polkit policies.
+type Specification struct {
+	policyFiles map[string]Policy
+}
+
+// AddPolicy adds a polkit policy file to install.
+func (spec *Specification) AddPolicy(name string, content Policy) error {
```

Comment:
(one plus of the approach here is that there's not chance of produce an unexpected name, which will be detected only at EnsureDirState time)

on the not great side, is that it's unclear what name means here exactly

Reply:
Yeah.  The fact that the systemd backend relies on the service names matching a particular pattern makes it feel a bit weird that it doesn't enforce that.  It's not at all clear that either of the interfaces using that backend care about the full systemd service name.